### PR TITLE
USACO Platinum Load Balancing Alternative Solution

### PR DESCRIPTION
**Pull request by dutinmeow**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-624.mdx [hunk 1]
```diff
@@ -2,21 +2,175 @@
...
+
+#define X first
+#define Y second
+
+const int MAXN = 1e5;
+
+int N;
+pair<int, int> A[MAXN];
+
+struct SegTree {
```
Comment (bqi343) [inline solutions/usaco-624.mdx:58]: this code doesn't even compile? confused

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/usaco-624.mdx [hunk 2]
```diff
@@ -2,21 +2,165 @@
...
+	int f(int l, int r) {
+		int res = 0;
+		for (l += N, r += N + 1; l < r; l >>= 1, r >>= 1) {
+			if (l & 1) res += T[l++];
+			if (r & 1) res += T[--r];
+		}
+		return res;
+	}
+
+	int eval(int x) {
```
[the remark this answers is not in the document]
Comment (dutinmeow) [reply solutions/usaco-624.mdx]: I use the term $f$ in the solution

File: solutions/usaco-624.mdx (hunk 1 above)
Comment (dutinmeow) [reply solutions/usaco-624.mdx:58]: oop, it should work now

File: solutions/usaco-624.mdx [hunk 3]
```diff
@@ -271,89 +403,94 @@ int main() {
...
-		br.close();
-	}
-
-	static class BIT {
-		public int[] bit;
-		public BIT(int N) {
-			bit = new int[N + 1];
-		}
+    public static void main(String[] args) throws IOException {
+    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-624.mdx:437]: the diff clearly shows that something has changed

[the remark this answers is not in the document]
Comment (dutinmeow) [reply solutions/usaco-624.mdx]: fixed

**Review by bqi343 — CHANGES_REQUESTED** [review]
Several things:
 - Complexity should be Nlog^2N
 - I just tried submitting this code and I got WA on the sample.
 - I'm not convinced that this is correct. You can't ternary search on functions of this form unless they are *strictly* decreasing, then *strictly* increasing.

**Comment by dutinmeow** [comment]
It was working fine before, I'll look into it

**Comment by dutinmeow** [comment]
ok found the issue

**Comment by bqi343** [comment]
> I'm not convinced that this is correct. You can't ternary search on functions of this form unless they are strictly decreasing, then strictly increasing.

For example, consider a function that is 1 everywhere except at one specific point. You can't detect this efficiently. Though it's probably hard to construct test cases that break this ...

Your code outputs 3 on this case, but the answer should be 2:

```
5
15 1
19 1
19 7
3 1
11 1
```

(Side note: The current solution and the official solution both fail N=1 :/)

**Comment by dutinmeow** [comment]
I don't think that's fixable if the ternary search itself is wrong?

**Comment by dutinmeow** [comment]
Maybe leave the solution up as a sort of extra note but with a warning box since it works on USACO, or just close it?

**Pull request closed without merging**